atlasdb-api/src/main/java/com/palantir/atlasdb/persistent/api/PersistentStore.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.persistent.api;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.annotation.Nonnull;
+
+import org.immutables.value.Value;
+
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+
+public interface PersistentStore<K, V> extends AutoCloseable {
+    @Value.Immutable
+    interface EntryFamilyHandle {
```

Comment:
I think it'll suffice to just call it `Handle`. Since we'll more often than not refer to it as `PersistentStore.Handle` which makes sense by itself.

Reply:
Renamed to handle and added docs explaining usage, creation and what does it represent.